vcap_utils/vcap_utils/backends/base_openvino.py

```diff
@@ -35,34 +35,112 @@ def __init__(self, model_xml: bytes,
+            if _RANGE_FOR_ASYNC_INFER_REQUESTS in supported_metrics:
+                low, high, _ = self.ie.get_metric(
+                    device_name, _RANGE_FOR_ASYNC_INFER_REQUESTS)
+                # Cap the num_requests, because setting it too high can crash
+                # TODO(Alex): Figure out _why_ hddl crashes when set to 'high'
+                num_requests = max(0, min(low * 2, high))
 
         self.net: IENetwork = self.ie.read_network(
             model=model_xml,
             weights=weights_bin,
             init_from_buffer=True)
 
-        self.exec_net: ExecutableNetwork = self.ie.load_network(
-            network=self.net,
-            device_name=device_name,
-            num_requests=n_requests)
+        try:
+            self.exec_net: ExecutableNetwork = self.ie.load_network(
+                network=self.net,
+                device_name=device_name,
+                num_requests=num_requests)
+        except RuntimeError as e:
+            if device_name == "CPU":
+                # It's unknown why this would happen when loading
+                # onto CPU, so we re-raise the error.
+                raise e
+
+            # This error happens when trying to load onto HDDL or MYRIAD, but
+            # somehow the device fails to work. In these cases, we try again
+            # but load onto 'CPU', which is a safe choice.
+            msg = f"Failed to load {self.__class__} onto device " \
+                  f"{device_name}. Error: '{e}'. " \
+                  f"Trying again with device_name='CPU'"
+            logging.critical(msg)
+            self.__init__(
+                model_xml=model_xml,
+                weights_bin=weights_bin,
+                device_name="CPU",
+                ie_core=ie_core)
+            return
 
         # Pull out a couple useful constants
-        self.InferRequestStatusCode = StatusCode
         self.input_blob_names: List[str] = list(self.net.inputs.keys())
         self.output_blob_names: List[str] = list(self.net.outputs.keys())
 
+        # For running threaded requests to the network
+        self._get_free_request_lock = RLock()
+        self._request_free_events: List[Event] = [
+            Event() for _ in self.exec_net.requests]
+        for request_free in self._request_free_events:
+            request_free.set()
+
+        # For keeping track of the workload for this backend
+        self._total_requests: int = len(self.exec_net.requests)
+        self._num_ongoing_requests: int = 0
+        self._num_ongoing_requests_lock: RLock = RLock()
+
+    @property
+    def workload(self) -> float:
+        """Returns the percent saturation of this backend. The backend is
+        at max 'efficiency' once the number of ongoing requests is equal to
+        or over number of exec_net.requests
+        """
+        return self._num_ongoing_requests / self._total_requests
+
+    def send_to_batch(self, input_data: OV_INPUT_TYPE) -> Queue:
```

Comment:
Okay! Here's the magical speed improvements. Here I override the send_to_batch method that would ordinarily send the image to the oven, and instead I send it directly to OpenVINO.